packages/@aws-cdk/aws-elasticloadbalancing/lib/load-balancer.ts

```diff
@@ -251,20 +251,21 @@ export class LoadBalancer extends Resource implements IConnectable {
 
   private readonly instancePorts: number[] = [];
   private readonly targets: ILoadBalancerTarget[] = [];
+  private readonly instanceIds: string[] = [];
 
   constructor(scope: Construct, id: string, props: LoadBalancerProps) {
     super(scope, id);
 
     this.securityGroup = new SecurityGroup(this, 'SecurityGroup', { vpc: props.vpc, allowAllOutbound: false });
     this.connections = new Connections({ securityGroups: [this.securityGroup] });
-
     // Depending on whether the ELB has public or internal IPs, pick the right backend subnets
     const selectedSubnets: SelectedSubnets = loadBalancerSubnets(props);
 
     this.elb = new CfnLoadBalancer(this, 'Resource', {
       securityGroups: [this.securityGroup.securityGroupId],
       subnets: selectedSubnets.subnetIds,
       listeners: Lazy.any({ produce: () => this.listeners }),
+      instances: Lazy.list({ produce: () => this.instanceIds.length == 0 ? undefined : this.instanceIds }),
```

Comment:
I *think* our linter will have you write triple equals (`===`) instead of double equals (`==`), does it not?

```suggestion
      instances: Lazy.list({ produce: () => this.instanceIds.length === 0 ? undefined : this.instanceIds }),
```

This is a crazy JavaScript thing. Read about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness).

Reply:
@pattasai Even better, I think we can use this [property](https://github.com/aws/aws-cdk/blob/d5a50c3cf7e55e4b8aa29524279e4af99050304a/packages/%40aws-cdk/core/lib/lazy.ts#L133) instead.

```ts
Lazy.list({ produce: () => this.instanceIds }, { omitEmptyArray: true });
```